pageserver/src/tenant/timeline/eviction_task.rs

```diff
@@ -282,5 +286,57 @@ impl Timeline {
                 );
             }
         }
+
+        // imitate the synthetic size calculation done by the consumption_metrics module.
+        // it runs as a per-tenant task, but the eviction_task.rs is per-timeline.
+        // so, some coordination is required.
+        if let Ok(tenant) = crate::tenant::mgr::get_tenant(self.tenant_id, true).await {
+            // The goal is to make the layer accesses that synthetic size calculation would do
+            // at a period slightly below `p.threshold`.
+            // We want them to happen at least every `p.threshold`, so that eviction won't evict them.
+            // We don't want them to happen more often, because they are somewhat expensive (CPU, IOs).
+            let mut g = tenant.eviction_task_synthetic_size.lock().await;
+            match *g {
+                Some(ts) if ts.elapsed() < p.period => { /* no need to run */ }
+                None | Some(_) => {
+                    let limit = {
+                        // continued horrible hacks; let's clone the global semaphore from pageserver conf as a
+                        // "a new" semaphore or create a new one for eviction tasks if the global has only one
+                        // permit.
+                        static LIMIT: sycn::OnceCell<Arc<sync::Semaphore>> = Default::default();
+                        let synthetic_size_task_semaphore =
+                            &self.conf.concurrent_tenant_size_logical_size_queries;
+                        if synthetic_size_task_semaphore.initial_permits().get() == 1 {
+                            LIMIT.get_or_init(|| Arc::new(sync::Semaphore::new(1)))
+                        } else {
+                            LIMIT.get_or_init(|| synthetic_size_task_semaphore.inner().clone())
+                        }
```

Comment:
Is this business even necessary? Why can't we just always use the `concurrent_tenant_size_logical_size_queries` ?

Reply:
Discussed and these are now duplicating the original semaphore as separate semaphore between evictions and original use. This is probably fine.